eva-accession-core/src/main/java/uk/ac/ebi/eva/accession/core/SubmittedVariantAccessioningService.java

```diff
@@ -59,24 +61,31 @@ public SubmittedVariantAccessioningService(SubmittedVariantMonotonicAccessioning
             throws AccessionCouldNotBeGeneratedException {
         List<AccessionWrapper<ISubmittedVariant, String, Long>> dbsnpVariants = accessioningServiceDbsnp.get(variants);
         List<ISubmittedVariant> variantsNotInDbsnp = removeFromList(variants, dbsnpVariants);
-        List<AccessionWrapper<ISubmittedVariant, String, Long>> submittedVariants = accessioningService.getOrCreate(
-                variantsNotInDbsnp);
-        return joinLists(submittedVariants, dbsnpVariants);
+        if (variantsNotInDbsnp.isEmpty()) {
+            // check this special case because mongo bulk inserts don't allow inserting empty lists
+            // (accession-commons BasicMongoDbAccessionedCustomRepositoryImpl.insert would need to change)
+            return dbsnpVariants;
+        } else {
+            List<AccessionWrapper<ISubmittedVariant, String, Long>> submittedVariants = accessioningService.getOrCreate(
+                    variantsNotInDbsnp);
+            return joinLists(submittedVariants, dbsnpVariants);
+        }
     }
 
     private List<ISubmittedVariant> removeFromList(List<? extends ISubmittedVariant> allVariants,
                                                    List<AccessionWrapper<ISubmittedVariant, String, Long>>
-                                                           vatiantsToDelete) {
-        return allVariants.stream().filter(variant -> !contains(vatiantsToDelete, variant))
+                                                           variantsToDelete) {
+        Set<String> hashesToDelete = variantsToDelete.stream()
+                                                     .map(AccessionWrapper::getHash)
+                                                     .collect(Collectors.toSet());
+
+        return allVariants.stream()
+                          .map(variant -> Pair.of(accessioningServiceDbsnp.getHash(variant), variant))
+                          .filter(pair -> !hashesToDelete.contains(pair.getFirst()))
+                          .map(Pair::getSecond)
```

Comment:
Is this to address [this issue](https://ebi-eva.slack.com/archives/C5A6MLDAR/p1562166268149400)?

Reply:
yes